### DISCUSSION TITLE
Generate HL7 messages from FHIR

### DISCUSSION THREAD

**Discussion by Hassan-Gaad** [nextgenhealthcare/connect#5126] [Q&A]
Hi there, hope someone helps in this **what I need is:** to generate HL7 messages from Openmrs to exchange them with another healthcare platform, **what do I know?** I am a frontend developer, and a newbie with Openmrs, FHIR, and no knowledge about the Mirth integration engine and HL7 **What do I have?** I have an openmrs instance running (inside Bahmni ) in a VM and the FHIR module is added and I can access the Patients object and other resources using the FHIR API, I am working with a certified Mirth connect integration engine engineer remotely, that requested from me to provide him with the patient's information of Openmrs to generate HL7 messages using Mirth integration engine and after I have asked on Openmrs forum how to get it, and why I need it? they suggested adding the FHIR module, and I did it, then I gave him the FHIR API details, and he can fetch the resources data successfully, but now he says that is complex to fetch data from different resources and he asked me to give the hl7 messages directly I don't know how? and is it possible to do so using Mirth? , I don't know exactly how mirth is working and what it needs, so any help, please?

**Comment by pacmano1** [comment]
So - a few possible reasons he or she wants you to generate HL7 might be:

1) The FHIR implementation is immature and does not support the required transactions for your integrations.   Although it seems since you said he/she can fetch the data which make his request a bit strange to me.  FHIR calls are just API calls after all.
2) They assume HL7 is more mature and has the transactional information required to do whatever you are trying to integrate without making many FHIR calls.
3) Since HL7v2 has been around so long, they may assume it is a light lift for you to make the EHR send messages.
4) They poorly scoped the work order thinking it would be easy.   Or if they are doing it for free are not prepared to do the amount of work.
5) They may not have FHIR experience so they perceive the work as hard.  
6) The other system already accepts HL7 so the integration is might be easier for them.

That is not a complete list, but you did not share the exact data you are trying to send.   "The patients information" is not detail enough.   In the design of an integration, one of the things you typically do is thoroughly describe the data needed to and from each party, what is that?   If you are transferring a lot of information about a patient that may in fact require dozens of calls.    

It is possible to transform FHIR to HL7v2 with an engine like Mirth, but depending on the data needed that is not trivial for an experience developer, let alone for someone with your current experience.   Given that fact, I would go back to your certified developer and change the scope (and thereby likely cost) to get them to do all the work in the transactional type your team is willing to pay for.   

Or of course asking the OpenMRS forum again about HL7 if in fact the transactions are supported in that module.  Although a quick look at the specification for that module seems to show the level of HL7 support in OpenMRS is very little, only ORU RO1 messages.

**Comment by siddharthsharma9258** [comment]
Great question and Great Answer!
